MyRangesImplementation/MyRangesImplementation/cartezian_product_iterator_test.cpp

```diff
@@ -0,0 +1,76 @@
+#include"cartezian_product_iterator.hpp"
+#include<catch.hpp>
+
+
+namespace Catch {
+	template<typename... Iterators>
+	struct StringMaker<ranges::internals::iterators::cartezian_product_iterator<Iterators...>> {
+		static std::string convert(const ranges::internals::iterators::cartezian_product_iterator<Iterators...>& iter) {
+			return StringMaker<decltype(*iter)>().convert(*iter);
+		}
+	};
+}
+
+TEST_CASE("checking cartezian_product_iterator functionallity", "[cartezian_product_iterator]") {
+	auto vec = std::vector{ 1,2 };
+	auto lst = std::list{ 3,4 };
+
+	auto vecProductLstIterator = ranges::internals::iterators::cartezian_product_iterator(
+		std::pair{ vec.begin(), vec.end() }, std::pair{ lst.begin(), lst.end() }
+	);
+
+	SECTION("check that given two iterators points to the begin of the {vec, lst} pair, operator == returns true", "[cartezian_product_iterator]") {
+		auto vecProductLstIterator2 = ranges::internals::iterators::cartezian_product_iterator(
+			std::pair{ vec.begin(), vec.end() }, std::pair{ lst.begin(), lst.end() }
+		);
+		REQUIRE(vecProductLstIterator == vecProductLstIterator2);
+	}
+
+	SECTION("check that given two different iterators operator = returns false", "[cartezian_product_iterator]") {
+		auto vecProductLstIterator2 = ranges::internals::iterators::cartezian_product_iterator(
+			std::pair{ ++vec.begin(), vec.end() }, std::pair{ lst.begin(), lst.end() }
+		);
+		REQUIRE_FALSE(vecProductLstIterator == vecProductLstIterator2);
+	}
+
+	SECTION("check that given two different iterators operator != returns true", "[cartezian_product_iterator]") {
+		auto vecProductLstIterator2 = ranges::internals::iterators::cartezian_product_iterator(
+			std::pair{ ++vec.begin(), vec.end() }, std::pair{ lst.begin(), lst.end() }
+		);
+		REQUIRE(vecProductLstIterator != vecProductLstIterator2);
+	}
+
+	SECTION("check that given two iterators points to the begin of the {vec, lst} pair, operator != returns false", "[cartezian_product_iterator]") {
+		auto vecProductLstIterator2 = ranges::internals::iterators::cartezian_product_iterator(
+			std::pair{ vec.begin(), vec.end() }, std::pair{ lst.begin(), lst.end() }
+		);
+		REQUIRE_FALSE(vecProductLstIterator != vecProductLstIterator2);
+	}
+
+	SECTION("check that operator * on vecProductLstIterator returns pair of {1, 3}", "[cartezian_product_iterator]") {
```

Comment:
the test name describes what it does.
consider choosing a name to explain what it tests.
I'll give you an example:
When calling operator *, expecting to get the first pair in the cartezian product